testing/acceptance_tests/circuit_test.py

```diff
@@ -0,0 +1,695 @@
+    ):
+        # Given
+        expected_parameter_grid_filename = "parameter-grid.json"
+
+        # When
+        build_uniform_param_grid(
+            number_of_params_per_layer=number_of_params_per_layer,
+            number_of_layers=number_of_layers,
+            min_value=min_value,
+            max_value=max_value,
+            step=step,
+        )
+
+        # Then
+        param_grid = load_parameter_grid(expected_parameter_grid_filename)
+        os.remove(expected_parameter_grid_filename)
+
+    def test_build_uniform_param_grid_fails_with_both_ansatz_specs_and_number_of_params_per_layer(
+        self,
+    ):
+        # Given
+        number_of_params_per_layer = 2
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": 2,
+            "problem_size": 1,
+        }
+
+        # When
+        with pytest.raises(AssertionError):
+            build_uniform_param_grid(
+                ansatz_specs=ansatz_specs,
+                number_of_params_per_layer=number_of_params_per_layer,
+            )
+
+    def test_build_uniform_param_grid_fails_with_neither_ansatz_specs_nor_number_of_params_per_layer(
+        self,
+    ):
+        # When
+        with pytest.raises(AssertionError):
+            build_uniform_param_grid()
+
+
+class Test_build_circuit_layers_and_connectivity:
+    @pytest.mark.parametrize(
+        "x_dimension, y_dimension, layer_type",
+        [
+            (0, None, "nearest-neighbor"),
+            (1, None, "nearest-neighbor"),
+            (2, None, "nearest-neighbor"),
+            (0, 0, "nearest-neighbor"),
+            (1, 0, "nearest-neighbor"),
+            (2, 0, "nearest-neighbor"),
+            (0, 1, "nearest-neighbor"),
+            (1, 1, "nearest-neighbor"),
+            (2, 1, "nearest-neighbor"),
+            (0, 2, "nearest-neighbor"),
+            (1, 2, "nearest-neighbor"),
+            (2, 2, "nearest-neighbor"),
+            (1, 1, "sycamore"),
+            (2, 1, "sycamore"),
+            (1, 2, "sycamore"),
+            (2, 2, "sycamore"),
+        ],
+    )
+    def test_build_circuit_layers_and_connectivity(
+        self, x_dimension, y_dimension, layer_type
+    ):
+        # Given
+        expected_circuit_layers_filename = "circuit-layers.json"
+        expected_circuit_connectivity_filename = "circuit-connectivity.json"
+
+        # When
+        build_circuit_layers_and_connectivity(
+            x_dimension=x_dimension, y_dimension=y_dimension, layer_type=layer_type
+        )
+
+        # Then
+        circuit_layers = load_circuit_layers(expected_circuit_layers_filename)
+        circuit_connectivity = load_circuit_connectivity(
+            expected_circuit_connectivity_filename
+        )
+
+        os.remove(expected_circuit_layers_filename)
+        os.remove(expected_circuit_connectivity_filename)
+
+
+class Test_create_random_circuit:
```

Comment:
All the class names should be CamelCase, right?

Reply:
Addressed here: 
f31c6d8